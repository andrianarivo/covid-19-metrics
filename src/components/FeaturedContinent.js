export default function FeaturedContinent() {
  return (
    <div>
      featured
    </div>
  );
}
