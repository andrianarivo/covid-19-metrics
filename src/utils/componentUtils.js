import asia from '../images/asia.png';
import africa from '../images/africa.png';
import northAmerica from '../images/north_america.png';
import southAmerica from '../images/south_america.png';
import oceania from '../images/oceania.png';
import europe from '../images/europe.png';

const getImage = (continent) => {
  switch (continent.toLowerCase()) {
    case 'asia':
      return asia;
    case 'africa':
      return africa;
    case 'north america':
      return northAmerica;
    case 'south america':
      return southAmerica;
    case 'australia-oceania':
      return oceania;
    case 'europe':
      return europe;
    default:
      return asia;
  }
};

export default getImage;
