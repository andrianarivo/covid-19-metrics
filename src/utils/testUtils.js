import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

export function withRouter(children) {
  return (
    <BrowserRouter>
      {children}
    </BrowserRouter>
  );
}

export function withProvider(children, mockStore = store) {
  return (
    <Provider store={mockStore}>
      {children}
    </Provider>
  );
}
