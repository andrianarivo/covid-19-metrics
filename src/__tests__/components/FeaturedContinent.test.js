import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { withRouter } from '../../utils/testUtils';
import FeaturedContinent from '../../components/FeaturedContinent';

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUseNavigate,
}));

describe('Testing FeaturedContinent component', () => {
  test('renders correctly', () => {
    const tree = create(withRouter(<FeaturedContinent
      continent="Dummy"
      active={0}
      updated={0}
    />));
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('clicks the button and navigate', async () => {
    // Arrange
    render(<FeaturedContinent continent="Dummy" active={0} updated={0} interactive />);

    // Act
    const button = screen.getByTestId('button-Dummy');
    await userEvent.click(button);

    // Assert
    expect(mockUseNavigate).toBeCalledWith('details/Dummy', {
      state: {
        active: 0,
        continent: 'Dummy',
        updated: 0,
      },
    });
    mockUseNavigate.mockRestore();
  });
});
