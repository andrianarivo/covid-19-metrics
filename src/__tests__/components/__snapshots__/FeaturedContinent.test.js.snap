// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FeaturedContinent component renders correctly 1`] = `
<div
  className="sc-jzuGiw kOlThV"
  image="Dummy"
>
  <div />
  <div>
    <p>
      Dummy
    </p>
    <p>
      <span>
        0
      </span>
       
      <span>
        active cases
      </span>
    </p>
    <p>
      updated
       53 years ago
    </p>
    <button
      data-testid="button-Dummy"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </div>
</div>
`;
